File: coding_py/bj18258.py
import sys
from collections import deque
input = sys.stdin.readline

n = int(input())
q = deque()

for _ in range(n):
    tmp = input().split()
    if tmp[0]=='push':
        q.append(int(tmp[1]))
    elif tmp[0]=='front':
        print(q[0] if q else -1)
    elif tmp[0]=='pop':
        print(q.popleft() if q else -1)
    elif tmp[0]=='size':
        print(len(q))
    elif tmp[0]=='empty':
        print(0 if q else 1)
    elif tmp[0]=='back':
        print(q[-1] if q else -1)

File: algo_cpp/bj18258.cpp
#include <iostream>
#include <queue>
#include <string>
using namespace std;

queue<int> q;
int main(void) {
	cin.tie(0); ios_base::sync_with_stdio(0);
	int n, tmp;
	string cmd;
	cin >> n;
	while (n--) {
		cin >> cmd;
		if (cmd=="push") {
			cin >> tmp;
			q.push(tmp);
		}
		else if (cmd == "front") {
			if (!q.empty())
				cout << q.front() << "\n";
			else
				cout << "-1\n";
		}
		else if (cmd == "back") {
			if (!q.empty())
				cout << q.back() << "\n";
			else
				cout << "-1\n";
		}
		else if (cmd == "size") {
			cout << q.size() << "\n";
		}
		else if (cmd == "empty") {
			if (!q.empty()) {
				cout << "0\n";
			}
			else
				cout << "1\n";
		}
		else if (cmd == "pop") {
			if (!q.empty()) {
				cout << q.front() << "\n";
				q.pop();
			}
			else
				cout << "-1\n";
		}
	}
	return 0;
}